### PR TITLE
test/fsx:  Remove the dead code associated with aio backend

### PR DESCRIPTION
**Pull request by johnzzpcrystal**

### CODE REVIEW

File: src/test/librbd/fsx.cc [hunk 1]
```diff
@@ -2504,8 +2504,10 @@ __aio_rw(int rw, int fd, char *buf, unsigned len, unsigned offset)
 	} while (ret == -EINTR);
 
 	if (ret != 1) {
-		if (ret == 0)
+		if (ret == 0) {
```
Comment (tchaikov) [inline src/test/librbd/fsx.cc]: please note that this is dead code in ceph. it's never executed in the ceph variant of fsx. instead, we use `{p,librbd_}{write,read}` for kbd/rbd-nbd and rbd modes. i wonder if we can just remove the aio backend for the read/write tests.

Comment (dillaman) [inline src/test/librbd/fsx.cc]: @tchaikov I'm fine w/ removing the dead code

Comment (johnzzpcrystal) [inline src/test/librbd/fsx.cc]: OK, I'll close it later

Comment (johnzzpcrystal) [inline src/test/librbd/fsx.cc]: Maybe I could upload a new patch to remove these dead code ?

Comment (dillaman) [inline src/test/librbd/fsx.cc]: @johnzzpcrystal Sounds good to me -- just force push your changes to the existing branch.

**Comment by dillaman** [comment]
@johnzzpcrystal Please squash your three commits down into a single commit (since the first two will essentially be dropped by the third).

**Comment by johnzzpcrystal** [comment]
@dillaman, Could this be merged ?

**Comment by tchaikov** [comment]
@johnzzpcrystal  i am running this PR through the rados test suite. but we might need to run it through the rbd suite also.

**Comment by johnzzpcrystal** [comment]
@tchaikov, Thanks a lot !

**Comment by dillaman** [comment]
@tchaikov Yes, it needs to go through an rbd suite run

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-05-06_17:11:36-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2017-05-09_10:05:25-rados-wip-kefu-testing---basic-mira/

**Comment by liewegas** [comment]
oops!  didn't see the rbd comment. hopefully this one is okay.

**Pull request merged**